docs/UserGuide.adoc

```diff
@@ -111,6 +116,27 @@ Deletes the 2nd person in the address book.
 `delete 1` +
 Deletes the 1st person in the results of the `find` command.
 
+== Deleting a person by name: `delete_name`
+
+Deletes the specified person from the address book after indexing other commands. Irreversible. +
+Format: `delete NAME`
+
+****
+Deletes the person by `NAME` not case sensitive.
+`NAME` refers to the full name of the person.
+****
+
+Examples:
+
+* `list` +
+`delete_name Matthew Alexander` +
+Deletes the Matthew Alexander in the address book.
+
+* `find Betsy` +
+`delete_name Betsy` +
+Deletes the 1st person in the results of the `find` command.
+
+
```

Comment:
You should create a clean PR, with only changes relevant to your new feature. So don't merge changes in the master branch into the branch for LO submission.